### ISSUE TITLE
Unexpected behavior of automatic locale selection

### ISSUE THREAD

**Issue by ezwelty** [falling-fruit/falling-fruit-web#656]
Reported via email by a user:

> This beta website is in French. I am not French and am not currently in France. I was in France like 2 months ago, and some other websites were in French at that time, but they have long since changed back to English. So I'm forced to conclude that there might be an issue with your website. I am using Chrome -- when I switch to Firefox, it's in English.

This may be hard to address or improve on, but sharing here in case we can do something about it. I suspect that since it worked differently in the two browsers, that maybe the user visited the beta website once in Chrome while in France and the automatic language choice stuck via local storage?

**Comment by wbazant** [comment]
I saw it as well after inviting someone to test the app a week ago -  their mobile picked the French locale for no discernible reason. They got on well enough, were stuck on an 'Elderberry' they wanted to add being missing, and I went to 'Settings' and changed it for them.

What it made me think, the language setting should be visible from the first interaction, in case it's wrong: on mobile it's one more reason to get a welcome screen (I've made #657 to track), and on desktop I think we should have the language setting in the top bar as on the live site.

> the automatic language choice stuck via local storage

I also wonder if we should only be preserving the language choice when logged in "Remember me" mode?

**Comment by wbazant** [comment]
Looked into this a bit:
>  if we should only be preserving the language choice when logged in "Remember me" mode?

wasn't a good idea - using sessionStorage is useless for language selection, because opening something in a new tab doesn't preserve the setting.

New proposed resolution is to only save the language when a user selects one through a dropdown, or upon logging in in 'remember me' mode, and to clear the setting on log out.

Something for the ideas basket: I've read it's possible to do a server side setting by IP - the server reads an IP, puts a tag into the HTML that the i18n language detector uses, and it's super-customised. Not sure how much I like it, following the browser setting seems more respectful of the user's intent!

**Comment by ezwelty** [comment]
> What it made me think, the language setting should be visible from the first interaction, in case it's wrong: on mobile it's one more reason to get a welcome screen (I've made [#657](https://github.com/falling-fruit/falling-fruit-web/issues/657) to track), and on desktop I think we should have the language setting in the top bar as on the live site.

Agreed. I'd been meaning to make an issue to suggest moving the language dropdown to the top bar.

> New proposed resolution is to only save the language when a user selects one through a dropdown, or upon logging in in 'remember me' mode, and to clear the setting on log out.

"save the language when a user selects one through a dropdown" makes a lot of sense to me. But why also "upon logging in in 'remember me' mode". Why should this be treated differently from other session types?

**Comment by wbazant** [comment]
> "save the language when a user selects one through a dropdown" makes a lot of sense to me. But why also "upon logging in in 'remember me' mode"?

I think I was trying to honour the 'remember me' opt-in a bit too much - no good reason. Then when implementing I ended up saving the language on log in regardless of 'remember me' preference, thinking that if someone successfully logged in the site is probably in a language they are able to use, but maybe there's no benefit.

My new proposed resolution is to save the language when a user selects one through a drop-down, and to clear it on log out.
